common/interfaces/user_interface/ui_input_text.c

```diff
@@ -548,8 +549,10 @@ static void next_btn_event_handler(lv_obj_t *next_btn, const lv_event_t event) {
       /* As of now, all text is stored in the buffer data->entered_text,
        * therefore we must copy data from here into the buffer which the
        * application can access */
-      ui_fill_text(
-          data->entered_text, data->input_text_ptr, data->max_input_size);
+      ui_fill_text(data->entered_text,
+                   data->input_text_ptr,
+                   data->max_input_size +
+                       1);    // Extra 1 byte for NULL character in size
```

Comment:
take the buffer size in the function parameter.

Reply:
As discussed, we should introduce this change in logic but because of lack of unit tests and additional time required to verify the change, added a buffer size variable to record the max buffer size for the string.